main.go

```diff
@@ -285,3 +318,37 @@ func main() { //nolint:funlen
 		os.Exit(1)
 	}
 }
+
+func setSecretCache(platform cluster.Platform) map[string]cache.Config {
```

Comment:
I have the same concern about naming. For my opinion it does not "set" anything, but creates the config. I saw in other projects that using "new" for such cases is idiomatic for golang as well, but not "set".

Reply:
createSecretCacheConfig() sounds clear for me